lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -835,6 +836,176 @@ class RAIIOpenedArchetypesTracker {
   }
 };
 
+/// Recursively traverse the conformance lists to determine sole conforming
+/// class, struct or enum type.
+NominalTypeDecl *
+SILCombiner::findSoleConformingTypeFromPCA(ProtocolDecl *Protocol) {
+  SmallVector<ProtocolDecl *, 16> PDWorkList;
```

Comment:
All of this except for the CHA part should go into the protocol conformance analysis